documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -1053,10 +1053,25 @@ Hibernate Reactive session associated with the context.
 === Vert.x instance service
 
 The `VertxInstance` service defines how Hibernate Reactive obtains an instance
-of Vert.x. The default implementation just creates one the first time it's
-needed. But if your program requires control over how the Vert.x instance is
-created, or how it's obtained, you can override the default implementation and
-provide your own `VertxInstance`. Let's consider this example:
+of Vert.x when there is no instance associated with the calling thread. The
+default implementation just creates one the first time it's needed. But if your
+program requires control over how the Vert.x instance is created, or how it's
+obtained, you can override the default implementation and provide your own
+`VertxInstance`.
+
+IMPORTANT: If your program starts Vert.x externally to Hibernate Reactive, and
+you don't tell Hibernate Reactive how to obtain this instance of Vert.x by
+specifying an implementation of `VertxInstance`, you might end up with multiple
+instances of `Vertx` in your program, resulting in confusing error messages.
```

Comment:
Isn't this a bug from our side? Shouldn't everything work anyway even if we start a different Vert.x instance?

Reply:
Well,  I would expect it to work if one uses `.withSession`.

Reply:
If the program already has a `Vertx`, and we go and create one, then that's simply wrong, I would have thought.

Reply:
I remember reading that one could use these scenario if they want different event busses for different groups of Verticles.
I might have dreamt it.
But it, even if it's unlikely, I'm not sure it's wrong